skimage/exposure/tests/test_exposure.py

```diff
@@ -267,6 +267,18 @@ def test_rescale_uint14_limits():
     assert_array_almost_equal(out, [0, uint14_max])
 
 
+def test_rescale_all_zeros():
+    image = np.zeros((2, 2), dtype=np.uint8)
+    out = exposure.rescale_intensity(image)
+    assert ~np.isnan(out).all()
```

Comment:
We know what the output shouldn't be. Do we know what the output should be?

Reply:
Yep, it should be the same

Reply:
it is probably because of the changes I requested. I think due to the logic that you have now, omin and omax are the same, and thus the image will be rescaled to zero. Not sure what the right behavior here is.